### PR TITLE
Matplotlib to chaco cheatsheet in docs 

### LINKED ISSUE #690: In documentation, consider providing a cheatsheet for Chaco users coming from Matplotlib
I know Chaco and Matplotlib have different mental models (and object models), but there's a handful of things that users are likely to do and for which they might know the Matplotlib API. I often share the following table with students and think it'd be nice to have something like it be part of the docs.

|          What          |           matplotlib          |                                 Chaco                                  |
|:----------------------:|:-----------------------------:|:----------------------------------------------------------------------:|
|       Plot lines       |        `ax.plot(x, y)`        |                          `p.plot(('x', 'y'))`                          |
|        x label         |    `ax.set_xlabel('Time')`    |                        `p.x_axis.title = 'Time'`                        |
|        y label         |    `ax.set_ylabel('Value')`   |                       `p.y_axis.title = 'Value'`                        |
|         Title          |     `ax.set_title(title)`     |                           `p.title = title`                            |
|        X limits        |   `ax.set_xlim((low, high))`  | `p.index_range.low_setting = LOW`  `p.index_range.high_setting = HIGH` |
|        Y limits        |   `ax.set_ylim((low, high))`  | `p.value_range.low_setting = LOW`  `p.value_range.high_setting = HIGH` |
| Label lines for legend |  `ax.plot(…, label=label)`  |                        `p.plot(…, name=name)`                        |
|       Add legend       |         `ax.legend()`         |                       `p.legend.visible = True`                        |
|    Legend position     | `ax.legend(loc='lower left')` |                      `p.legend_alignment = 'll'`                       |
|    Background color    |                               |                         ` p.bgcolor = 'white'`                         |
|      Show spines       |                               |                       `p.border_visible = True`                        |
|       Line width       |    `ax.plot(linewidth=lw)`    |          `p.line_width = 1.1` or `p.plot(…, linewidth=lw)`           |
|         X grid         |      `ax.grid(axis='x')`      |                       `p.x_axis.visible = True`                        |
|         Y grid         |      ` ax.grid(axis='y')`     |                       `p.y_axis.visible = True`                        |
|     Second y ticks     |                               |                  `line.y_axis.orientation = "right"`                   |

It's not 100% "complete" and it may need some context before it to explain what `ax`, `p`, and `line` are, but it's a start.

### PR DESCRIPTION
**Pull request by aaronayres35**
closes #690 
This PR adds a brief section to the documentation giving a "cheatsheet" of code "equivalences" between matplotlib and chaco.

It effectively copies the table given in the issue, and adds a short introduction to it / fills in a couple empty sections

### CODE REVIEW

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+As mentioned in the :ref:`Chaco Tutorial <tutorial>`,
+`matplotlib <https://matplotlib.org/>`_ and Chaco have different
+approaches to plotting (script-oriented vs. application oriented). However, new
+users of chaco who have worked with matplotlib in the past may find it useful
+to think about mapping "equivalent" code between the two.
+
+The following table lists various plotting related operations one might want to
+perform along with code for doing that using either Chaco or Matplotlib. For
+the Matplotlib code ``ax`` repesents the Axes that what would result from
+running  ``fig, ax = plt.subplots()`` after importing
```
Comment (rahulporuri) [inline docs/source/user_manual/matplotlib_to_chaco.rst:15]:
```suggestion
the Matplotlib code ``ax`` repesents the Plot Axes
i.e. ``fig, ax = plt.subplots()`` after importing
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+``import matplotlib.pyplot as plt``. For more details see the
+`matplotlib User's Guide <https://matplotlib.org/stable/users/index.html>`_.
+For Chaco code, ``p`` is the :class:`~.Plot` instance.
+
++------------------------+------------------------------------------+------------------------------------------------------------------------+
+|          What          |                matplotlib                |                                 Chaco                                  |
++========================+==========================================+========================================================================+
+|       Plot lines       |             `ax.plot(x, y)`              |                          `p.plot(('x', 'y'))`                          |
```
Comment (rahulporuri) [inline docs/source/user_manual/matplotlib_to_chaco.rst:23]: nit - there's unnecessary whitespace in the "Chaco" column - can you remove that?

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+|       Line width       |         `ax.plot(linewidth=lw)`          |           `p.line_width = 1.1` or `p.plot(…, linewidth=lw)`            |
++------------------------+------------------------------------------+------------------------------------------------------------------------+
+|         X grid         |           `ax.grid(axis='x')`            |                       `p.x_axis.visible = True`                        |
++------------------------+------------------------------------------+------------------------------------------------------------------------+
+|         Y grid         |           `ax.grid(axis='y')`            |                       `p.y_axis.visible = True`                        |
++------------------------+------------------------------------------+------------------------------------------------------------------------+
+|     Second y ticks     |  `ax.yaxis.set_label_position('right')`  |                   `p.y_axis.orientation = "right"`                     |
+|                        |         `ax.yaxis.tick_right()`          |                                                                        |
```
Comment (rahulporuri) [inline docs/source/user_manual/matplotlib_to_chaco.rst:54]: having two entries in the "matplotlib" column here seems to mess with the formatting - the first entry is in bold as if it were a title.

File: docs/source/user_manual/matplotlib_to_chaco.rst (hunk 2 above)
Comment (rahulporuri) [inline docs/source/user_manual/matplotlib_to_chaco.rst:23]: it's not straightforward for people to know what the string `'x'` and `'y'` mean. It would be useful if you expanded on the earlier paragraph by adding some more information on instantiating the `Plot` object with the traits `x` and `y`, which we are referring to here.

**Review by rahulporuri — COMMENTED** [review]
Mostly LGTM

File: docs/source/user_manual/matplotlib_to_chaco.rst (hunk 3 above)
Comment (aaronayres35) [reply docs/source/user_manual/matplotlib_to_chaco.rst:54]: Any idea how to remedy this?

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+to think about mapping "equivalent" code between the two.
+
+The following table lists various plotting related operations one might want to
+perform along with code for doing that using either Chaco or Matplotlib. For
+the Matplotlib code ``ax`` repesents the Plot Axes
+i.e. ``fig, ax = plt.subplots()`` after importing
+``import matplotlib.pyplot as plt``. For more details see the
+`matplotlib User's Guide <https://matplotlib.org/stable/users/index.html>`_.
+For Chaco code, ``p`` is the :class:`~.Plot` instance.  ``x`` and ``y`` are the
+data to be plotted. In chaco, this data needs to be wrapped in some
```
Comment (rahulporuri) [inline docs/source/user_manual/matplotlib_to_chaco.rst:19]:
```suggestion
data to be plotted. In chaco, this data needs to be wrapped in a
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    pd.set_data("x", x)
+    pd.set_data("y", y)
+    p = Plot(pd)
+
+Note that in the matplotlib column, ``x`` and ``y`` represent the data, but
+``'x'`` and ``'y'``, such as in the "X grid" and "Y Grid" rows are simply
+options for the ``axis`` argument. In chaco ``'x'`` and ``'y'`` are the names
+the ``PlotData`` object associates with numpy arrays ``x`` and ``y``.
+
+.. list-table:: Matplotlib To Chaco Cheatsheet
```
Comment (rahulporuri) [inline docs/source/user_manual/matplotlib_to_chaco.rst:39]: there's something new i learnt today!

**Review by rahulporuri — APPROVED** [review]
LGTM with one minor nitpicky comment

**Comment by rahulporuri** [comment]
@achabotl can you make time to review this PR?

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
+.. _matplotlib2chaco:
+
+##############################
+Matplotlib To Chaco Cheatsheet
+##############################
+As mentioned in the :ref:`Chaco Tutorial <tutorial>`,
+`matplotlib <https://matplotlib.org/>`_ and Chaco have different
+approaches to plotting (script-oriented vs. application-oriented). However, new
+users of chaco who have worked with matplotlib in the past may find it useful
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:9]:
```suggestion
users of Chaco who have worked with matplotlib in the past may find it useful
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
+.. _matplotlib2chaco:
+
+##############################
+Matplotlib To Chaco Cheatsheet
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:4]:
```suggestion
Matplotlib to Chaco Cheatsheet
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+##############################
+As mentioned in the :ref:`Chaco Tutorial <tutorial>`,
+`matplotlib <https://matplotlib.org/>`_ and Chaco have different
+approaches to plotting (script-oriented vs. application-oriented). However, new
+users of chaco who have worked with matplotlib in the past may find it useful
+to think about mapping "equivalent" code between the two.
+
+The following table lists various plotting related operations one might want to
+perform along with code for doing that using either Chaco or Matplotlib. For
+the Matplotlib code ``ax`` repesents the Plot Axes
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:14]:
```suggestion
perform along with code for doing that using either Chaco or matplotlib. For
the matplotlib code ``ax`` represents the Plot Axes
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+i.e. ``fig, ax = plt.subplots()`` after importing
+``import matplotlib.pyplot as plt``. For more details see the
+`matplotlib User's Guide <https://matplotlib.org/stable/users/index.html>`_.
+For Chaco code, ``p`` is the :class:`~.Plot` instance.  ``x`` and ``y`` are the
+data to be plotted. In chaco, this data needs to be wrapped in a
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:19]:
```suggestion
data to be plotted. In Chaco, this data needs to be wrapped in a
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+``PlotData`` object. For example,
+
+::
+
+    import numpy as np
+    x = np.linspace(-2*pi, 2*pi, 100)
+    y = sin(x)
+
+    # chaco only
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:28]:
```suggestion
    # Chaco only
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    pd = ArrayPlotData()
+    pd.set_data("x", x)
+    pd.set_data("y", y)
+    p = Plot(pd)
+
+Note that in the matplotlib column, ``x`` and ``y`` represent the data, but
+``'x'`` and ``'y'``, such as in the "X grid" and "Y Grid" rows are simply
+options for the ``axis`` argument. In chaco ``'x'`` and ``'y'`` are the names
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:36]:
```suggestion
options for the ``axis`` argument. In Chaco ``'x'`` and ``'y'`` are the names
```

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+      - `p.y_axis.title = 'Value'`
+    * - Title
+      - `ax.set_title(title)`
+      - `p.title = title`
+    * - X limits
+      - `ax.set_xlim((low, high))`
+      - `p.index_range.low_setting = LOW` `p.index_range.high_setting = HIGH`
+    * - Y limits
+      - `ax.set_ylim((low, high))`
+      - `p.value_range.low_setting = LOW` `p.value_range.high_setting = HIGH`
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:63]: Use lower case for low and high, like in the rest of the table.

File: docs/source/user_manual/matplotlib_to_chaco.rst [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+.. list-table:: Matplotlib To Chaco Cheatsheet
+    :widths: 25 40 50
+    :header-rows: 1
+
+    * - What
+      - Matplotlib
+      - Chaco
+    * - Plot lines
+      - `ax.plot(x, y)`
+      - `p.plot(('x', 'y'))`
```
Comment (achabotl) [inline docs/source/user_manual/matplotlib_to_chaco.rst:48]: Needs double backticks to render as code:

```suggestion
      - ``ax.plot(x, y)``
      - ``p.plot(('x', 'y'))``
```

**Review by achabotl — COMMENTED** [review]
Thank you for making this! LGTM.

See inline comments, mainly for consistency of naming and some formatting.
